eth-contracts/contracts/Governance.sol

```diff
@@ -296,6 +316,70 @@ contract Governance is RegistryContract {
         return outcome;
     }
 
+    function vetoProposal(uint256 _proposalId) external {
+        requireIsInitialized();
+
+        require(msg.sender == guardianAddress, "Governance::vetoProposal:Only guardian can veto proposals.");
+
+        require(
+            _proposalId <= lastProposalId && _proposalId > 0,
+            "Governance::vetoProposal:Must provide valid non-zero _proposalId."
+        );
+
+        require(
+            proposals[_proposalId].outcome == Outcome.InProgress,
+            "Governance::vetoProposal:Cannot veto inactive proposal."
+        );
+
+        proposals[_proposalId].outcome = Outcome.No;
+
+        emit ProposalVetoed(_proposalId);
+    }
+
+    // ========================================= Guardian Actions =========================================
+
+    /**
+    TODO
+    - take contractKey instead and validate that key points to valid registered contract
+    - figure out function + event naming
+    - do we need txHash in internal _execTx?
+    */
+    function guardianExecuteTransaction(
+        address _targetContractAddress,
+        uint256 _callValue,
+        string calldata _signature,
+        bytes calldata _callData
+    ) external
+    {
```

Comment:
why have a separate function like this instead of having _executeTranasction accept these parameters like previously? this is fine functionally but we have duplicated the below code now. if we had _executeTransaction accept these parameters you could keep the new event and such (which is good) but have zero dupe'd code